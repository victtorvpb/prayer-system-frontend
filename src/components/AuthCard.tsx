import { Paper } from "@mui/material";
export function AuthCard({ children }: { children: React.ReactNode }) {
  return (
    <Paper
      elevation={6}
      sx={{
        p: 5,
        width: 370,
        borderRadius: 4,
        display: "flex",
        flexDirection: "column",
        alignItems: "center",
        position: "relative",
        zIndex: 1,
        background: "rgba(255,255,255,0.75)",
        backdropFilter: "blur(8px)",
        boxShadow: "0 8px 32px 0 rgba(31, 38, 135, 0.15)",
      }}
    >
      {children}
    </Paper>
  );
}
