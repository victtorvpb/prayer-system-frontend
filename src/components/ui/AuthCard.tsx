import { Paper } from "@mui/material";
export function AuthCard({ children }: { children: React.ReactNode }) {
  return (
    <Paper
      elevation={0}
      sx={{
        p: { xs: 3, sm: 5 },
        width: { xs: "100%", sm: 380 },
        maxWidth: "95vw",
        borderRadius: 5,
        display: "flex",
        flexDirection: "column",
        alignItems: "center",
        position: "relative",
        zIndex: 1,
        background: "rgba(255,255,255,0.85)",
        backdropFilter: "blur(12px)",
        boxShadow: "0 8px 32px 0 rgba(25, 118, 210, 0.12)",
      }}
    >
      {children}
    </Paper>
  );
}
