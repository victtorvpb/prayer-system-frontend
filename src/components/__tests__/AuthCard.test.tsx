import { render, screen } from "@testing-library/react";
import { AuthCard } from "../AuthCard";

describe("AuthCard", () => {
  it("renderiza o conteúdo passado como children", () => {
    render(
      <AuthCard>
        <div>Conteúdo de teste</div>
      </AuthCard>
    );
    expect(screen.getByText("Conteúdo de teste")).toBeInTheDocument();
  });

  it("bate o snapshot do AuthCard", () => {
    const { asFragment } = render(
      <AuthCard>
        <div>Snapshot test</div>
      </AuthCard>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
