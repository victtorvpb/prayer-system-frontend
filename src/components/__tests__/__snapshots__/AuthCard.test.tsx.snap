// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthCard bate o snapshot do AuthCard 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 css-1g1z0vk-MuiPaper-root"
    style="--Paper-shadow: none;"
  >
    <div>
      Snapshot test
    </div>
  </div>
</DocumentFragment>
`;
